### PR TITLE
Rename $IsOSX to $IsMacOS

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
PS-Committee decided we should be consistent with our naming and conform to Apple's use of macOS instead of OSX, however, for readability and consistently we are staying with Pascal casing.

Also renamed instances of variations of `OSX` to `macOS` for consistency in comments/text (separate commit to make it easier to review)

Fix https://github.com/PowerShell/PowerShell/issues/4700

### CODE REVIEW

**Comment by daxian-dbw** [comment]
I don't see changes to `build.psm1`, is that file missed from this PR?

**Comment by SteveL-MSFT** [comment]
@daxian-dbw not sure how I missed that, will fix

**Comment by iSazonov** [comment]
[Apple Inc. ](https://www.apple.com/lae/macos/sierra/) uses `macOS` not `MacOS` - maybe we should use the same too? - `$IsmacOS`

**Comment by SteveL-MSFT** [comment]
@iSazonov we had that discussion in the issue, the consensus is that $IsMacOS is preferred over $IsmacOS for readability and other languages (like Java) use isMacOS, so it's already somewhat common.

File: tools/travis.ps1 [hunk 1]
```diff
@@ -42,7 +42,7 @@ function Set-DailyBuildBadge
     $storageAccountKey = $Env:TestResultAccountKey
 
     # this is the url referenced in README.MD which displays the badge
-    $platform = if ( $IsOSX ) { "OSX" } else { "Linux" }
+    $platform = if ( $IsLinux ) { "Linux" } else { "OSX" }
```
Comment (iSazonov) [inline tools/travis.ps1]: Minor comment - maybe replace "OSX" too?

**Review by iSazonov — COMMENTED** [review]
LGTM

File: tools/travis.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply tools/travis.ps1]: The file stored in Azure for the badge has OSX in the filename.  I'm fine leaving that one.

**Review by daxian-dbw — APPROVED** [review]
The package name for macOS should also be changed, but that can be a separate PR.

**Comment by markekraus** [comment]
Should this run through the CI with [feature] to ensure none of the feature tests broke with the change?

**Comment by SteveL-MSFT** [comment]
@markekraus good point, I'll resubmit

**Comment by SteveL-MSFT** [comment]
AppVeyor failure is due to the current issue causing nightly to fail and @dantraMSFT is looking into it

**Comment by dantraMSFT** [comment]
Yes, the failure is due to https://github.com/PowerShell/PowerShell/pull/4763. You can ignore it.

**Comment by SteveL-MSFT** [comment]
Rebased to pick up fix

**Review by iSazonov — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**